src/libm_helper.rs

```diff
@@ -0,0 +1,143 @@
+use core::marker::PhantomData;
+
+use crate::*;
+
+pub struct Libm<T>(PhantomData<T>);
```

Comment:
Add some doc comments here to explain what this type is and some examples of how it can be used.